app/models/application_choice.rb

```diff
@@ -69,20 +69,45 @@ def course_full_error
     I18n.t('errors.application_choices.course_full', descriptor: course.provider_and_name_code)
   end
 
-  def chosen_site_full?
+  def course_option_full?
```

Comment:
Thinking about the API for application_choices, and wondering if there's potentially a way to simplify.

We have two clients for this collection of availability checking code - the `ApplicationFormPresenter` and the `CourseChoicesReviewComponent`. In both cases, they need to be aware of an implicit order in which to query these methods, based on increasing specificity of what's being checked.

What if we had:

```
def site_full?
  course.course_options.where(site: course_option.site).vacancies.blank?
end
```

```
def study_mode_full?
  course_option.no_vacancies?
end
```
We then evaluate them in that order. So if we reach `study_mode_full?`, then we know that there are vacancies at the same site in the other study mode. The specs for these methods will probably be simpler/easier to follow as well. 

The fact that there's a hierarchy to these method calls at all means we should consider extracting an object out that deals exclusively with the availability errors on a given application choice, but that's something for another time.